### PR TITLE
Allow model migrations to use the model DB

### PR DESCRIPTION
**Pull request by SimonRichardson**
This refactors the facades to hand out model exporters and importers. It hides the controller and model DB, preventing the pollution of very low level types to facades. This should prevent any facade code from interacting with a database directly. We're explicitly calling out that they should either use a service or a model migration factory.

This was a long held design issue that we've always wanted to clean up. I believe this is probably the best approach; it provides better guard rails.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
$ juju bootstrap lxd src --build-agent
$ juju add-model moveme
$ juju deploy ubuntu

$ juju bootstrap lxd dst --build-agent
$ juju switch src
$ juju migrate moveme dst
```

## Links

**Jira card:** JUJU-5325

### CODE REVIEW

File: apiserver/facades/controller/migrationmaster/facade.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ import (
 
 // ModelExporter exports a model to a description.Model.
 type ModelExporter interface {
-	ExportModel(ctx context.Context, leaders map[string]string, store objectstore.ObjectStore) (description.Model, error)
+	ExportModel(context.Context, map[string]string, objectstore.ObjectStore) (description.Model, error)
```
Comment (manadart) [inline apiserver/facades/controller/migrationmaster/facade.go:32]: Comment this if you are removing the names.

File: internal/migration/migration.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ type ModelExporter struct {
 }
 
 func NewModelExporter(legacyStateExporter LegacyStateExporter, scope modelmigration.Scope) *ModelExporter {
```
Comment (manadart) [inline internal/migration/migration.go:52]: Comment please.

There's also a typo: "dataase" on line 79.

**Review by manadart — APPROVED** [review]
This is a really great clean-up. 

Smashing to get rid of direct controller DB access via the facade context.

Note on QA (not an issue introduced here): I see this:
```
machine-0: 18:43:54 ERROR juju.worker.dependency "migration-master" manifold worker returned unexpected error: failed to set phase: could not get migration: model "6c4bb443-6759-4851-8e68-f1c68c6414b6" not found (not found)
```
I've seen it before. I believe this causes us not to link the migrated model with the external controller, meaning we don't get a redirect.

Previously, status would tell the user where the model now was, but now it just says not found.

**Comment by SimonRichardson** [comment]
@manadart I'll set aside some time next pulse to go over migrations. I've got to fix a few issues that are beginning to pile up as we're adding more code.

 - Passing `nil` into the migrations services for the watchers. This was brought up by @tlm and needs fixing before it becomes a massive problem later down the line.
 - Not passing a logger everywhere is causing some pain points.
 - I'll fix this bug you noticed, whilst I'm doing that.

**Pull request merged**